site/schemas/disruption.schema.ts

```diff
@@ -25,3 +27,54 @@ export const disruptionSchema = createDisruptionsSchemaRefined.and(
 );
 
 export type Disruption = z.infer<typeof disruptionSchema>;
+
+export const exportFileSchema = z.object({
+    exportType: z.union(
+        [z.literal("csv"), z.literal("excel"), z.literal("pdf")],
+        setZodDefaultError("Select a format to export"),
+    ),
+});
+
+export type ExportFileType = z.infer<typeof exportFileSchema>;
+
+const displayValidityPeriod = z.object({
+    startTime: z.string(),
+    endTime: z.string().optional().nullish(),
+});
+
+export const exportDisruptionsSchema = z.array(
+    z
+        .object({
+            index: z.number(),
+            summary: z.string(),
+            modes: z.array(z.string()),
+            isOperatorWideCq: z.boolean(),
+            isNetworkWideCq: z.boolean(),
+            serviceIds: z.array(z.string()).optional(),
+            stopsAffectedCount: z.number(),
+            validityPeriods: z.array(displayValidityPeriod),
+            severity: z.string(),
+            isLive: z.boolean(),
+            status: z.string(),
+        })
+        .transform((item) => {
+            return {
+                id: item.index,
+                title: item.summary,
+                mode: item.modes.map((mode) => splitCamelCaseToString(mode)).join(", ") || "N/A",
+                "operator wide": item.isOperatorWideCq ? "yes" : "no",
+                "network wide": item.isNetworkWideCq ? "yes" : "no",
+                "services affected": item.serviceIds ? item.serviceIds.length : 0,
+                "stops affected": item.stopsAffectedCount,
```

Comment:
In the ticket the titles of the csv are in camel case like stopsAffected

Reply:
done after checking with chris